localstack/aws/protocol/parser.py

```diff
@@ -509,30 +509,54 @@ def __init__(self, service: ServiceModel) -> None:
         self.operation_lookup: DefaultDict[
             str, DefaultDict[Pattern[str], OperationModel]
         ] = defaultdict(lambda: defaultdict())
```

Comment:
Should this be a `defaultdict(lambda: OrderedDict())`, to preserve the order of inserts when iterating over `sorted_operation_models` below?

Reply:
Good catch! I wasn't aware of the fact that some version of Python do not preserve the order of inserts. However, CPython 3.6 and [Python 3.7+](https://docs.python.org/3.7/library/stdtypes.html#dictionary-view-objects) preserve the order of inserts on standard dict:
> Changed in version 3.7: Dictionary order is guaranteed to be insertion order.

Should I change it to OrderedDict to support non-CPython 3.6?
Since Python 3.6 is maintained anymore, we could also think about dropping Python 3.6 support in localstack?

Reply:
Good point - I think we can drop <3.6 (at least soon-ish, perhaps with a pre-announcement in the next release), but in general I think it would be good to make it explicit in the code if we rely on ordering (could also be through a comment that briefly explains it). 👍

Reply:
I also agree that it should definitely be as explicit as possible. I pushed a commit with your suggested changes.
Thanks!